src/vdaf/prio3.rs

```diff
@@ -854,7 +840,7 @@ where
 /// Message sent by the [`Client`] to each [`Aggregator`] during the Sharding phase.
 #[derive(Clone, Debug)]
 pub struct Prio3InputShare<F, const SEED_SIZE: usize> {
-    /// The measurement share.
+    /// The share for measurement and proof
     measurement_share: Share<F, SEED_SIZE>,
 
     /// The proof share.
```

Comment:
The serialization format for helper input shares is either one seed (without joint randomness) or two (with joint randomness), so the `Encode` and `Decode` implementations will need updating too.

Another idea: we could change this to the following:

```rust
pub enum Prio3InputShare<F, const SEED_SIZE: usize> {
    Leader {
        measurement_share: Vec<F>,
        proofs_share: Vec<F>,
        joint_rand_blind: Option<Seed<SEED_SIZE>>,
    },
    Helper {
        meas_and_proofs_share: Seed<SEED_SIZE>,
        joint_rand_blind: Option<Seed<SEED_SIZE>>,
    },
}
```

This would eliminate the redundancy in the helper share.

Reply:
The new commit makes this enum. I didn't modify how the encoding works. Let me know if I should change that. There's no enum tag necessary because it's given in the decoding params.